// mui-autocomplete/dist/action.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("@material-ui/core/styles");

var _CircularProgress = _interopRequireDefault(require("@material-ui/core/CircularProgress"));

var _Cancel = _interopRequireDefault(require("@material-ui/icons/Cancel"));

var _Search = _interopRequireDefault(require("@material-ui/icons/Search"));

var _IconButton = _interopRequireDefault(require("@material-ui/core/IconButton"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    icon: {
      padding: 2
    }
  };
});

function Action(props) {
  var classes = useStyles();

  switch (props.type) {
    case 'loading':
      return _react["default"].createElement(_CircularProgress["default"], {
        size: 20
      });

    case 'complete':
      return _react["default"].createElement(_IconButton["default"], {
        className: classes.icon,
        onClick: props.onClear
      }, _react["default"].createElement(_Cancel["default"], null));

    case 'search':
      return _react["default"].createElement(_IconButton["default"], {
        className: classes.icon
      }, _react["default"].createElement(_Search["default"], null));

    default:
      return _react["default"].createElement(_CircularProgress["default"], {
        size: 20
      });
  }
}

var _default = Action;
exports["default"] = _default;
